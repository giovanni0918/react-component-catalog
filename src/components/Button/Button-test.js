import React, { Component } from 'react';
import Button from './Button';
import renderer from 'react-test-renderer';

describe('Button', () => {
  it('should always render a button', () => {
    const buttons = lockScreen().find("button");
    expect(buttons.length).toBeGreaterThan(0);
  });
});