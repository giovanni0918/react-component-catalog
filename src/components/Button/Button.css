.Button {
    box-sizing: border-box;
    box-shadow: 0 2px 4px rgba(0, 0, 0, 0.3);
    min-width: 5rem;
    margin: 0;
    padding: 0.5rem;
    border: 1px solid #404040;
    border-radius: 2px;
    outline-color: transparent;
    font-family: Arial, Helvetica, sans-serif;
    font-size: 1rem;
    font-weight: 500;
    background-color: ghostwhite;
    color: black;
    text-transform: capitalize;
    will-change: transform;
    transform: scale(1.0);
    transition: transform 0.2s ease-in-out;
}

.Button:hover,
.Button:focus {
    box-shadow: 0 3px 5px rgba(0, 0, 0, 0.4);
}

.Button:active {
    transform: scale(1.1);
}

.Button--default {
    background-color: grey;
    color: black;
}

.Button--primary {
    background-color: dodgerblue;
    color: black;
}

.Button--info {
    background-color: powderblue;
    color: black;
}

.Button--success {
    background-color: springgreen;
    color: black;
}

.Button--warning {
    background-color: lightgoldenrodyellow;
    color: black;
}

.Button--danger {
    background-color: palevioletred;
    color: black;
}