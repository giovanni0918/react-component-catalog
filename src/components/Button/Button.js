// @flow
import React from 'react';
import styles from './Button.css';

type Props = { className: string, onClick: Function, textContent: string };

const Button = ({
  className,
  onClick,
  textContent
}: Props) => (
    <button
      className={className}
      onClick={onClick}>
      {textContent}
    </button>
  );

export default Button;