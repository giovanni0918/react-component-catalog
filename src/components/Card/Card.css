.Card {
    box-sizing: border-box;
    box-shadow: 0 3px 5px rgba(0, 0, 0, 0.4);
    min-height: 192px;
    max-width: 768px;
    margin: 0;
    padding: 1rem;
    font-family: Arial, Helvetica, sans-serif;
    font-size: 1rem;
    font-weight: 500;
    background-color: white;
    color: #202020;
}