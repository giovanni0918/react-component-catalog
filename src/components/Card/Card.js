// @flow
import React from 'react';
import styles from './Card.css';

type Props = { className: string, heading: string, copy: string, children: Array<Object> };

const Card = ({
  className,
  heading,
  copy,
  children
}: Props) => (
    <div className={className}>
      <h3>
        {heading}
      </h3>
      <p>
        {copy}
      </p>
      {children}
    </div>
  );
export default Card;