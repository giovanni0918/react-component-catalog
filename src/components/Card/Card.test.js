import React from 'react';
import Card from './Card';
import Button from '../Button/Button.js';
import renderer from 'react-test-renderer';

const SampleCard = (
  <Card
    className='Card'
    heading='Google Pixel'
    copy='Copper mug pork belly skateboard, freegan echo park fixie stumptown. Celiac banh mi tumblr fanny pack, intelligentsia live-edge glossier skill. Fanny pack pinterest celiac raw denim. Venmo master cleanse seitan williamsburg fab, mixtape sriracha synth lyft.'>
    <Button
      className='Button Button--primary'
      onClick={() => console.log('button inside a card')}
      textContent='button inside a card'
    />
  </Card>
);

test('It should always render a div', () => {
  const component = renderer.create(SampleCard);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('It should have a prop children of type=array', () => {
  const component = renderer.create(SampleCard);
  let tree = component.toJSON();
  expect(Array.isArray(tree.children)).toBe(true);
  expect(tree.children.length).toBeGreaterThanOrEqual(2);
});