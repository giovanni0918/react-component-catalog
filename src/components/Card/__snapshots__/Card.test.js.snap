// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It should always render a div 1`] = `
<div
  className="Card"
>
  <h3>
    Google Pixel
  </h3>
  <p>
    Copper mug pork belly skateboard, freegan echo park fixie stumptown. Celiac banh mi tumblr fanny pack, intelligentsia live-edge glossier skill. Fanny pack pinterest celiac raw denim. Venmo master cleanse seitan williamsburg fab, mixtape sriracha synth lyft.
  </p>
  <button
    className="Button Button--primary"
    onClick={[Function]}
  >
    button inside a card
  </button>
</div>
`;
